### DISCUSSION TITLE
Statistics for teachers

### DISCUSSION THREAD

**Discussion by boryanagoncharenko** [hedyorg/hedy#1241] [Implementation discussions]
We want to improve the teacher's interface by adding statistics which help teachers understand how their class is progressing. There already was a brainstorming session (#665) and the following stats were shortlisted:

- Runs per level (granularity: month, all time)
- Highest level reached per student (maximum)
- Error rate per student (granularity: week)
- Exceptions per level and per type (granularity: month, all time)

@OnnoEbbens @MarleenGilsing your input is needed! Are the above above useful? May be there are other metrics which would be better?

**Comment by OnnoEbbens** [comment]
@boryanagoncharenko I like the idea of including metrics. It is hard to tell which metrics contain the most information about student performance and how to help them. So I think it is nice to test the ones you mention and evaluate after some months. Maybe we can add the quiz scores per level.

**Reply by boryanagoncharenko** [reply]
> @boryanagoncharenko I like the idea of including metrics. It is hard to tell which metrics contain the most information about student performance and how to help them. So I think it is nice to test the ones you mention and evaluate after some months. Maybe we can add the quiz scores per level.

@OnnoEbbens our preliminary tests showed that metrics are indeed helpful for the admin, so I hope teachers will find the class metrics useful too.

What are interesting quiz scores for you? I was thinking that when selecting a given student to see:
- a chart with the successful / failed / abandoned attempts per level
- a chart with the total number of correctly and incorrectly answered questions per level (that is the submitted answers for all attempts)
What do you think?

**Reply by OnnoEbbens** [reply]
Thanks for looking into this. Both are interesting for me. If possible it would be nice to see this in one view for all students in a class. Then I can compare the students and who is struggling.

**Comment by boryanagoncharenko** [comment]
@OnnoEbbens @MarleenGilsing You should now be able to access to statistics information of your classes. To see the available stats, please chose "For teachers" from the Hedy menu and then select a class. On the class page, you should see a "Show class statistics" button.

Please note that quiz information is not available yet. Any feedback/comments/suggestions are appreciated.

**Reply by MarleenGilsing** [reply]
@boryanagoncharenko Maybe a dumb question, but what is your PR/branch called, because I can't seem to find it to test it?

**Reply by boryanagoncharenko** [reply]
@MarleenGilsing Since displaying real data in the charts might be critical for your experience and feedback, there is no branch. The feature is already on hedycode.com.

**Reply by MarleenGilsing** [reply]
Oh, I'm sorry I was looking for the button. But I guess you meant these statistics:
![image](https://user-images.githubusercontent.com/80678586/152336807-2ba8d18b-c5e7-4003-aae8-92acc07a0e5b.png)

They work great! 👍

**Reply by MarleenGilsing** [reply]
Only feedback I can think of is maybe making the Username list alphabetical?

**Reply by boryanagoncharenko** [reply]
No, these are not the statistics I am referring to. You should see a button right below with caption "Show class statistics":
![Screenshot 2022-02-01 at 17 29 38](https://user-images.githubusercontent.com/3010723/152337808-86803242-de0d-4e48-be4c-b9ace0f8fe97.png)

Let me know if you don't see it because I might have not whitelisted you correctly.

**Reply by MarleenGilsing** [reply]
I do see it now. Not sure if I was just blind before or my computer didn't show it yet, but that's fixed! I'll take a look!
